### PR TITLE
Giovanni/ui-fixes

### PR DESCRIPTION
**Pull request by sinkingsugar**
WIP

To fix all dangerous uses of `cloneVar`
Input -> Field and mutability

### CODE REVIEW

File: rust/src/types.rs [hunk 1]
```diff
@@ -3166,6 +3166,11 @@ impl ParamVar {
     }
   }
 
+  pub fn set_cloning(&mut self, value: &Var) {
```
Comment (guusw) [inline rust/src/types.rs:3169]: Rename this to set, and the other one to some set_unsafe, set_fast, set_direct or something. Mostly always it should be cloned

Comment (guusw) [reply rust/src/types.rs:3169]: Ideally we can have the fast path implemented in a reusable way in rust as well, and fall back to the cloneVar FFI, but for now renaming is fine.

**Review by guusw — CHANGES_REQUESTED** [review]

File: docs/samples/shards/UI/TextInput/1.edn [hunk 2]
```diff
@@ -11,6 +11,6 @@
    (UI.CentralPanel
     (->
      (Setup "Lorem ipsum dolor sit amet" >= .text)
-     (UI.TextInput :Variable .text :Multiline false))))
+     (UI.TextField :Variable .text :Multiline false))))
```
Comment (Kryptos-FR) [inline docs/samples/shards/UI/TextInput/1.edn:14]: File must also be renamed.

File: docs/samples/shards/UI/TextInput/2.edn [hunk 3]
```diff
@@ -11,6 +11,6 @@
    (UI.CentralPanel
     (->
        ;; read-only
-     (UI.TextInput "Lorem ipsum dolor sit amet"))))
+     (UI.TextField "Lorem ipsum dolor sit amet"))))
```
Comment (Kryptos-FR) [inline docs/samples/shards/UI/TextInput/2.edn:14]: File must also be renamed.

File: docs/samples/shards/UI/TextInput/3.edn [hunk 4]
```diff
@@ -11,6 +11,6 @@
    (UI.CentralPanel
     (->
      ;; exposing a new variable
-     (UI.TextInput .text))))
+     (UI.TextField .text))))
```
Comment (Kryptos-FR) [inline docs/samples/shards/UI/TextInput/3.edn:14]: File must also be renamed.

**Pull request merged**